library/core/src/iter/adapters/zip.rs

```diff
@@ -434,7 +434,7 @@ impl<A: Debug + TrustedRandomAccess, B: Debug + TrustedRandomAccess> ZipFmt<A, B
 ///    called on `self`:
 ///     * `std::clone::Clone::clone()`
 ///     * `std::iter::Iterator::size_hint()`
-///     * `std::iter::Iterator::next_back()`
+///     * `std::iter::DoubleEndedIterator::next_back()`
 ///     * `std::iter::Iterator::__iterator_get_unchecked()`
 ///     * `std::iter::TrustedRandomAccess::size()`
 ///
```

Comment:
I wonder why `TrustedRandomAccess` is in this file but not in a common file?

Reply:
It’s `use`’d elsewhere for implementing the `TrustedRandomAccess` trait itself, but AFAICT, `TrustedRandomAccess` is – at least currently – only necessary, i.e. practically used, to optimize `Zip`. (And of course it’s also “used” to _implement_ `__iterator_get_unchecked`.)

Reply:
One thing that I’m not sure of is why exactly `__iterator_get_unchecked` is not a method of `TrustedRandomAccess`. But I didn’t do the research yet trying to figure the reason behind this decision either. Also all of this discussion seems slightly off-topic for this PR that’s just about fixing the typo ;-)

Reply:
> But I didn’t do the research yet trying to figure the reason behind this decision either. 

Ah, here’s a [short] explanation https://github.com/rust-lang/rust/pull/73565#issuecomment-757535493

Reply:
Btw, TIL, there _is_ another use case. https://github.com/rust-lang/rust/issues/85873#issuecomment-853958658